simply/actor.py

```diff
@@ -844,3 +916,92 @@ def clip_soc(soc_prediction, upper_clipping):
         # clipping everything before local maximum
         soc_prediction[:idc_loc_max][soc_prediction[:idc_loc_max] > upper_clipping] = upper_clipping
         soc_max = np.max(soc_prediction)
+
+
+def get_price(pricing_strategy, index, final_price, energy):
+    """ Returns the price at the current time step to generate an order early
+
+    :param pricing_strategy: strategy name and parameters or function with arguments index, price
+        and energy amount
+    :type pricing_strategy: dict() or function
+    :param index: amount of time step until interaction with market maker is planned
+    :param final_price: price of the market maker interaction
+    :param energy: amount of energy that is planned for trading. Positive amounts are buying and
+        negative amounts are selling energy
+    :return: current price for order generation
+    :rtype: float
+    """
+    # the market schedule demands to buy or sell power in the current time slot. Therefore
+    # pricing will be adjusted to market maker prices, i.e. the final price will be used.
+    if index == 0:
+        return final_price
+
+    # a function can be given with the arguments index, final_price and energy. it should return
+    # a price. if even more advanced functions with access to the actor data shall be used the
+    # get_price method can be over written
+    if callable(pricing_strategy):
+        return pricing_strategy(index, final_price, energy)
+
+    if not isinstance(pricing_strategy, type(dict())):
+        raise TypeError("pricing_strategy is neither a callable function nor a dictionary.")
+
+    # pricing strategy can be linear
+    if pricing_strategy["name"] == "linear":
+        try:
+            m = pricing_strategy["param"][0]
+        except TypeError:
+            m = pricing_strategy["param"]
+        sign = np.sign(energy)
+        return final_price - sign * index * m * final_price
+
+    # pricing strategy can be harmonic
+    # harmonic pricing changes the price according to the harmonic series meaning
+    # 1, 1/2, 1/3, 1/4, 1/5 ... and so on.
+    if pricing_strategy["name"] == "harmonic":
+        sign = np.sign(energy)
+        # the half_life_index is the index where the price is 50% of the final price for buys.
+        # if energy is supposed to be sold, its 200% instead
```

Comment:
add sth. like: ..., not to be confused with an half-life time of an exponential decay function